src/aktualizr_repo/repo.h

```diff
@@ -14,24 +14,62 @@ struct KeyPair {
   std::string private_key;
 };
 
+struct Delegation {
+  Delegation() = default;
+  Delegation(const boost::filesystem::path &repo_path, std::string delegation_name) : name(std::move(delegation_name)) {
+    boost::filesystem::path delegation_path(((repo_path / "repo/image") / name).string() + ".json");
+    boost::filesystem::path targets_path(repo_path / "repo/image/targets.json");
+    if (!boost::filesystem::exists(delegation_path) || !boost::filesystem::exists(targets_path)) {
+      throw std::runtime_error(std::string("delegation ") + name + "does not exists");
+    }
+    Json::Value delegations = Utils::parseJSONFile(targets_path)["signed"]["delegations"];
+    for (const auto &role : delegations["roles"]) {
+      if (role["name"].asString() == name) {
+        pattern = role["paths"][0].asString();
+      }
+    }
+
+    if (pattern.empty()) {
+      throw std::runtime_error("Could not found delegation role in targets.json");
+    }
+  }
+  bool isMatched(const boost::filesystem::path &image_path) {
+    (void)image_path;
+    glob_t globbuf;
+    glob(pattern.c_str(), 0, nullptr, &globbuf);
```

Comment:
Three things:
1. What pulls in `glob_t`? I think a header file is missing.
2. I think we were trying to remove globbing from aktualizr because we didn't really need it, it is platform-specific code, and it wasn't available for Android. That said, aktualizr-repo is a host-side tool, so we are probably free to use it here.
3. Why the `(void)image_path` line?